File: Tests/DuetCLITests/Resources/mini-swift/src-ios/Subtrees/Counter/CounterFeature/Sources/CounterFeature/CounterFeature.swift
// Mini-tree source: the Swift-only counter feature (single implementation —
// no declaration parity applies; existence and geometry are what the lint pins).

public struct CounterState: Equatable {
  public var count: Int
}

public enum CounterAction: Equatable {
  case increment
  case reset
}

public enum CounterEffectPayload: Equatable {
  case announce(Int)
}

File: Tests/DuetCLITests/Resources/mini-kmp/src-kmp/subtrees/counter/logic/src/commonMain/kotlin/com/mini/counter/CounterFeature.kt
// Mini-tree source: the single-source counter feature (commonMain — the ONE
// implementation; existence and subtree geometry are what the lint pins).
package com.mini.counter

data class CounterState(
    val count: Int,
)

sealed interface CounterAction {
    data object Increment : CounterAction
    data object Reset : CounterAction
}

sealed interface CounterEffectPayload {
    data class Announce(val count: Int) : CounterEffectPayload
}
